test/tests/kernels/simple_transient_diffusion/tests

```diff
@@ -39,6 +39,7 @@
     requirement = "We shall be able to solve an initially poorly scaled problem by using MOOSE's automatic scaling feature in parallel"
     cli_args = 'Executioner/automatic_scaling=true -pc_type hypre -pc_hypre_type boomeramg'
     prereq = 'automatic-scaling-done-once'
+    max_parallel = 2 # Can't reset preallocation in PETSc when there are no dofs on a process
```

Comment:
@fdkong Do you think this is something reasonable from PETSc? After changing the communicator for this error (see https://bitbucket.org/petsc/petsc/pull-requests/1835/change-the-communicator-for/diff), I get:
```
[2]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[2]PETSC ERROR: Null argument, when expecting valid pointer
[2]PETSC ERROR: No saved preallocation info 

[2]PETSC ERROR: See http://www.mcs.anl.gov/petsc/documentation/faq.html for trouble shooting.
[2]PETSC ERROR: Petsc Release Version 3.11.2, unknown 
[2]PETSC ERROR: [3]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[3]PETSC ERROR: Null argument, when expecting valid pointer
[3]PETSC ERROR: No saved preallocation info 

[3]PETSC ERROR: See http://www.mcs.anl.gov/petsc/documentation/faq.html for trouble shooting.
[3]PETSC ERROR: Petsc Release Version 3.11.2, unknown 
[3]PETSC ERROR: ../../../moose_test-dbg on a arch-moose named Alexanders-Mac-Pro.local by lindad Mon Jul  1 14:22:08 2019
[3]PETSC ERROR: Configure options --download-hypre=1 --with-ssl=0 --with-debugging=no --with-pic=1 --with-shared-libraries=1 --with-cc=mpicc --with-cxx=mpicxx --with-fc=mpif90 --download-fblaslapack=1 --download-metis=1 --download-ptscotch=1 --download-parmetis=1 --download-superlu_dist=1 --download-mumps=1 --download-scalapack=1 --CC=mpicc --CXX=mpicxx --FC=mpif90 --F77=mpif77 --F90=mpif90 --CFLAGS="-fPIC -fopenmp" --CXXFLAGS="-fPIC -fopenmp" --FFLAGS="-fPIC -fopenmp" --FCFLAGS="-fPIC -fopenmp" --F90FLAGS="-fPIC -fopenmp" --F77FLAGS="-fPIC -fopenmp" --with-cxx-dialect=C++11 --with-fortran-bindings=0 --with-sowing=0
[3]PETSC ERROR: #1 MatResetPreallocation_SeqAIJ() line 3911 in /Users/lindad/projects/moose/petsc/src/mat/impls/aij/seq/aij.c
[3]PETSC ERROR: #2 MatResetPreallocation() line 847 in /Users/lindad/projects/moose/petsc/src/mat/interface/matrix.c
[3]PETSC ERROR: #3 MatResetPreallocation_MPIAIJ() line 2821 in /Users/lindad/projects/moose/petsc/src/mat/impls/aij/mpi/mpiaij.c
[3]PETSC ERROR: ../../../moose_test-dbg on a arch-moose named Alexanders-Mac-Pro.local by lindad Mon Jul  1 14:22:08 2019
#4 MatResetPreallocation() line 847 in /Users/lindad/projects/moose/petsc/src/mat/interface/matrix.c
[2]PETSC ERROR: Configure options --download-hypre=1 --with-ssl=0 --with-debugging=no --with-pic=1 --with-shared-libraries=1 --with-cc=mpicc --with-cxx=mpicxx --with-fc=mpif90 --download-fblaslapack=1 --download-metis=1 --download-ptscotch=1 --download-parmetis=1 --download-superlu_dist=1 --download-mumps=1 --download-scalapack=1 --CC=mpicc --CXX=mpicxx --FC=mpif90 --F77=mpif77 --F90=mpif90 --CFLAGS="-fPIC -fopenmp" --CXXFLAGS="-fPIC -fopenmp" --FFLAGS="-fPIC -fopenmp" --FCFLAGS="-fPIC -fopenmp" --F90FLAGS="-fPIC -fopenmp" --F77FLAGS="-fPIC -fopenmp" --with-cxx-dialect=C++11 --with-fortran-bindings=0 --with-sowing=0
[2]PETSC ERROR: #1 MatResetPreallocation_SeqAIJ() line 3911 in /Users/lindad/projects/moose/petsc/src/mat/impls/aij/seq/aij.c
[2]PETSC ERROR: #2 MatResetPreallocation() line 847 in /Users/lindad/projects/moose/petsc/src/mat/interface/matrix.c
[2]PETSC ERROR: #3 MatResetPreallocation_MPIAIJ() line 2821 in /Users/lindad/projects/moose/petsc/src/mat/impls/aij/mpi/mpiaij.c
[2]PETSC ERROR: #4 MatResetPreallocation() line 847 in /Users/lindad/projects/moose/petsc/src/mat/interface/matrix.c
```
when trying to run the above test case with 4 processes. This can be attributed to the fact that only two processes actually have dofs.

Reply:
I see. I know what is going on. I will fix PETSc soon.  

But at this point, let us do some restrictions on moose if autoscaling is used. For example, we should have at least one dof per core. 

Sorry